core/state/pruner/pruner.go

```diff
@@ -891,37 +895,53 @@ func asyncPruneExpiredStorageInDisk(diskdb ethdb.Database, pruneExpiredInDisk ch
 		log.Debug("found expired state", "addr", info.Addr, "path",
 			hex.EncodeToString(info.Path), "epoch", info.Epoch, "isBranch",
 			info.IsBranch, "isLeaf", info.IsLeaf)
+		itemCount++
 		addr := info.Addr
-		// delete trie kv
-		trieCount++
 		switch scheme {
 		case rawdb.PathScheme:
 			val := rawdb.ReadTrieNode(diskdb, addr, info.Path, info.Hash, rawdb.PathScheme)
-			trieSize += common.StorageSize(len(val) + 33 + len(info.Path))
-			rawdb.DeleteTrieNode(batch, addr, info.Path, info.Hash, rawdb.PathScheme)
+			if len(val) == 0 {
+				log.Debug("cannot find source trie?", "addr", addr, "path", info.Path, "hash", info.Hash, "epoch", info.Epoch)
+			} else {
+				trieCount++
+				trieSize += common.StorageSize(len(val) + 33 + len(info.Path))
+				rawdb.DeleteTrieNode(batch, addr, info.Path, info.Hash, rawdb.PathScheme)
+			}
```

Comment:
1. It may be better to do `if val == nil` instead of `if len(val) == 0`. `val` could be nil when the node key does not match the one in the db or there's an reading the value from the disk. 

2. It may be helpful to specify what does the 33 includes. Perhaps adding a constant variable or add comments.

Reply:
1. `len(val)==0` if better than `val == nil`, currenttly disk err will ignore;
2. will add more comments;